docs/source/migration-guides.rst

```diff
@@ -0,0 +1,258 @@
+    npmPath: "...",
+    compiler: "...",
+    docker-tag: "...",
+    solcVersion: "...",
+    legacyOutput: "...",
+    allowedPaths: "...",
+    ganacheOptions: "..."
+  }
+
+
+*Waffle 2.4.0*
+
+.. code-block:: text
+
+  {
+    sourceDirectory: "...",
+    outputDirectory: "...",
+    nodeModulesDirectory: "...",
+    compilerType: "...",
+    compilerVersion: "...",
+    compilerAllowedPaths: "..."
+  }
+
+Migration from Waffle 2.5.* to Waffle 3.0.0
+-------------------------------------------
+
+There are some new functionality and some slight refactoring and improved paradigms in Waffle v3.
+
+**Removed deprecated APIs from provider**
+
+In :code:`Waffle 3.0.0` we remove deprecated APIs from provider.
+
+**Swapped arguments for Fixture**
+
+In :code:`Waffle 3.0.0` we swapped arguments for Fixture, because the provider argument is very rarely used compared to wallets.
+So such implantation should be more convenient for users.
+
+*Waffle 2.5.0*
+
+.. code-block:: ts
+
+  import {expect} from 'chai';
+  import {loadFixture, deployContract} from 'ethereum-waffle';
+  import BasicTokenMock from './build/BasicTokenMock';
+
+  describe('Fixtures', () => {
+    async function fixture(provider, [wallet, other]) {
+      const token = await deployContract(wallet, BasicTokenMock, [
+        wallet.address, 1000
+      ]);
+      return {token, wallet, other};
+    }
+
+    it('Assigns initial balance', async () => {
+      const {token, wallet} = await loadFixture(fixture);
+      expect(await token.balanceOf(wallet.address)).to.equal(1000);
+    });
+
+    it('Transfer adds amount to destination account', async () => {
+      const {token, other} = await loadFixture(fixture);
+      await token.transfer(other.address, 7);
+      expect(await token.balanceOf(other.address)).to.equal(7);
+    });
+  });
+
+*Waffle 3.0.0*
+
+.. code-block:: ts
+
+  import {expect} from 'chai';
+  import {loadFixture, deployContract} from 'ethereum-waffle';
+  import BasicTokenMock from './build/BasicTokenMock';
+
+  describe('Fixtures', () => {
+    async function fixture([wallet, other], provider) {
+      const token = await deployContract(wallet, BasicTokenMock, [
+        wallet.address, 1000
+      ]);
+      return {token, wallet, other};
+    }
+
+    it('Assigns initial balance', async () => {
+      const {token, wallet} = await loadFixture(fixture);
+      expect(await token.balanceOf(wallet.address)).to.equal(1000);
+    });
+
+    it('Transfer adds amount to destination account', async () => {
+      const {token, other} = await loadFixture(fixture);
+      await token.transfer(other.address, 7);
+      expect(await token.balanceOf(other.address)).to.equal(7);
+    });
+  });
+
+**Added automatic recognising waffle.json config without cli argument**
+
+Waffle recognises :code:`waffle.json` as default configuration file. If your configuration file is called
+:code:`waffle.json`, it's possible to use just :code:`waffle` to build contracts.
```

Comment:
This is great, but it does not describe what the previous behaviour was if you did not provide the argument to the cli